Fixed build against OpenSSL 1.1

xvitaly opened the pull request:
Fixed build against OpenSSL 1.1. Patch backported from https://github.com/grishka/libtgvoip/pull/31 pull request.

Original patch by: Nicholas Guriev <guriev-ns@ya.ru>

xvitaly committed:
Fixed build against OpenSSL 1.1.

Patch backported from https://github.com/grishka/libtgvoip/pull/31 pull request.
Original patch by: Nicholas Guriev <guriev-ns@ya.ru>

Closes #43.

berkus reviewed (changes requested):
1. Rename original commit, move "patch backported..." etc lines to the body from subject

2. Split actual patch and error typo fixing into two commits (they are not related)

xvitaly committed:
Fixed spelling in libtgvoip third-party library.

Patch backported from https://github.com/grishka/libtgvoip/pull/31 pull request.
Original patch by: Nicholas Guriev <guriev-ns@ya.ru>

xvitaly commented:
Fixed.
